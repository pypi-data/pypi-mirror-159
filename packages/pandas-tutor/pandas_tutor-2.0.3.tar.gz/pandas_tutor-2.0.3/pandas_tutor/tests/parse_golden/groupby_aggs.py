(dogs
 .groupby('grooming')
 .agg('mean')
)

(dogs
 .groupby('grooming')
 .var()
)

# should not be parsed into an AggCall
(dogs
 .groupby('grooming')
 .transform(lambda x: x.max() - x.min())
)
