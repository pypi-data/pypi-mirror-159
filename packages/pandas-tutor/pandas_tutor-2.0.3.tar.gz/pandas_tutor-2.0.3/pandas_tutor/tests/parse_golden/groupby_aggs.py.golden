{
  "type_": "ParsedModule",
  "code": "(dogs\n .groupby('grooming')\n .agg('mean')\n)\n\n(dogs\n .groupby('grooming')\n .var()\n)\n\n# should not be parsed into an AggCall\n(dogs\n .groupby('grooming')\n .transform(lambda x: x.max() - x.min())\n)\n",
  "location": {
    "start": {
      "line": 0,
      "ch": 0
    },
    "end": {
      "line": 15,
      "ch": 0
    }
  },
  "statements": [
    {
      "type_": "ChainStatement",
      "code": "(dogs\n .groupby('grooming')\n .agg('mean')\n)",
      "location": {
        "start": {
          "line": 0,
          "ch": 0
        },
        "end": {
          "line": 3,
          "ch": 1
        }
      },
      "chain": [
        {
          "type_": "StartOfChain",
          "code": "dogs",
          "location": {
            "start": {
              "line": 0,
              "ch": 1
            },
            "end": {
              "line": 0,
              "ch": 5
            }
          }
        },
        {
          "type_": "GroupByCall",
          "code": "dogs\n .groupby('grooming')",
          "location": {
            "start": {
              "line": 1,
              "ch": 1
            },
            "end": {
              "line": 1,
              "ch": 21
            }
          },
          "axis": "index"
        },
        {
          "type_": "AggCall",
          "code": "(dogs\n .groupby('grooming')\n .agg('mean')\n)",
          "location": {
            "start": {
              "line": 2,
              "ch": 1
            },
            "end": {
              "line": 2,
              "ch": 13
            }
          }
        }
      ]
    },
    {
      "type_": "ChainStatement",
      "code": "(dogs\n .groupby('grooming')\n .var()\n)",
      "location": {
        "start": {
          "line": 5,
          "ch": 0
        },
        "end": {
          "line": 8,
          "ch": 1
        }
      },
      "chain": [
        {
          "type_": "StartOfChain",
          "code": "dogs",
          "location": {
            "start": {
              "line": 5,
              "ch": 1
            },
            "end": {
              "line": 5,
              "ch": 5
            }
          }
        },
        {
          "type_": "GroupByCall",
          "code": "dogs\n .groupby('grooming')",
          "location": {
            "start": {
              "line": 6,
              "ch": 1
            },
            "end": {
              "line": 6,
              "ch": 21
            }
          },
          "axis": "index"
        },
        {
          "type_": "AggCall",
          "code": "(dogs\n .groupby('grooming')\n .var()\n)",
          "location": {
            "start": {
              "line": 7,
              "ch": 1
            },
            "end": {
              "line": 7,
              "ch": 7
            }
          }
        }
      ]
    },
    {
      "type_": "ChainStatement",
      "code": "(dogs\n .groupby('grooming')\n .transform(lambda x: x.max() - x.min())\n)",
      "location": {
        "start": {
          "line": 11,
          "ch": 0
        },
        "end": {
          "line": 14,
          "ch": 1
        }
      },
      "chain": [
        {
          "type_": "StartOfChain",
          "code": "dogs",
          "location": {
            "start": {
              "line": 11,
              "ch": 1
            },
            "end": {
              "line": 11,
              "ch": 5
            }
          }
        },
        {
          "type_": "GroupByCall",
          "code": "dogs\n .groupby('grooming')",
          "location": {
            "start": {
              "line": 12,
              "ch": 1
            },
            "end": {
              "line": 12,
              "ch": 21
            }
          },
          "axis": "index"
        },
        {
          "type_": "PassThroughCall",
          "code": "(dogs\n .groupby('grooming')\n .transform(lambda x: x.max() - x.min())\n)",
          "location": {
            "start": {
              "line": 13,
              "ch": 1
            },
            "end": {
              "line": 13,
              "ch": 40
            }
          },
          "func": "transform"
        }
      ]
    }
  ]
}